alacritty/src/display/mod.rs

```diff
@@ -607,21 +509,50 @@ impl Display {
             ime: Ime::new(),
             highlighted_hint: None,
             vi_highlighted_hint: None,
-            #[cfg(not(any(target_os = "macos", windows)))]
-            is_x11,
+            is_wayland,
             cursor_hidden: false,
             visual_bell: VisualBell::from(&config.bell),
             colors: List::from(&config.colors),
             pending_update: Default::default(),
             pending_renderer_update: Default::default(),
-            is_damage_supported,
             debug_damage,
             damage_rects,
             next_frame_damage_rects,
             hint_mouse_point: None,
         })
     }
 
+    #[inline]
+    pub fn gl_context(&self) -> &PossiblyCurrentContext {
+        self.context.get()
+    }
+
+    pub fn make_not_current(&mut self) {
+        if self.context.get().is_current() {
+            self.context.replace_with(|context| {
+                context.make_not_current().expect("failed to disable context").treat_as_current()
+            });
+        }
+    }
+
+    pub fn make_current(&self) {
+        self.context.make_current(&self.surface).expect("failed to make context current.")
```

Comment:
We have a check in `make_not_current` to ensure it is current, but we have no check in here to ensure it is not current. Seems a bit strange I'd say unless we have a good reason for this. Otherwise I'd prefer consistency.